lib/atlas/dataset/derived_dataset.rb

```diff
@@ -0,0 +1,63 @@
+module Atlas
+  class Dataset::DerivedDataset < Dataset
```

Comment:
Yes! Very much in favour of this!
The current names date back to when we did not have a proper class hierarchy for this